presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -352,7 +352,15 @@ public PlanNode visitSample(SampleNode node, RewriteContext<Void> context)
         @Override
         public PlanNode visitValues(ValuesNode node, RewriteContext<Void> context)
         {
-            return context.defaultRewrite(node);
+            List<List<Expression>> canonicalizedRows = node.getRows().stream()
+                    .map(this::canonicalize)
+                    .collect(toImmutableList());
+            List<Symbol> canonicalizedOutputSymbols = canonicalizeAndDistinct(node.getOutputSymbols());
+            checkState(node.getOutputSymbols().size() == canonicalizedOutputSymbols.size(), "Values output symbols were pruned");
```

Comment:
just use `canonicalize` (if it exists) instead of `canonicalizeAndDistinct`

Reply:
No such method. If I use canonicalize then output symbols could silently have duplicates. Even though I use `canonicalizeAndDistinct` I test if none of the symbols were pruned. 

I would prefer to catch this query with the `checkState` to better understand the nature of the query instead of assuming something blindly here.